### ISSUE TITLE
Add user as a collaborator with permission parameter

### ISSUE THREAD

**Issue by snadon** [octokit/octokit.net#1101]
Should we be able to set permissions when adding a collaborator to a repository?

The GitHub API supports it:
https://developer.github.com/v3/repos/collaborators/#add-user-as-a-collaborator

But Octokit.Net does not:
https://github.com/octokit/octokit.net/blob/master/Octokit%2FClients%2FIRepoCollaboratorsClient.cs#L44

**Comment by Haacked** [comment]
Yeah, that looks like an oversight on our part. It think we should add an overload that accepts an enum with the available permission options.

**Comment by Haacked** [comment]
Want to submit a PR?

**Comment by M-Zuber** [comment]
And here I was talking bad about you.
![image](https://cloud.githubusercontent.com/assets/4746537/12930800/15d71fc8-cf83-11e5-8110-19b580e11c1f.png)
Please accept my apologies :smile:

**Comment by Haacked** [comment]
> Please accept my apologies

LOLOLOL! No apologies necessary. You're usually correct. I happen to be taking a break to try and respond to all the OSS projects I maintain so the timing was lucky.

**Comment by ryangribble** [comment]
Just a question on this out of interest... 

> I think we should add an overload that accepts an enum with the available permission options.

Are you saying @Haacked that this body parameter should actually be explicitly in the overloaded function, or should it be what appears to be the "usual" Octokit way where a Request object is passed in?

Just trying to get a feel for the octokit.net conventions, most I had come across seem to use the request object approach, so was surprised to see you mention adding an overload directly taking the enum paramter (unless i misunderstood)

Enum directly in Ad function
```
enum CollaboratorPermission { Read, Write, Admin }

Task Add(string owner, string repo, string user, CollaboratorPermission permission);
```

vs

Using a request object
```
enum CollaboratorPermission { Read, Write, Admin }

public class CollaboratorUpdate
{
   public CollaboratorPermission Permission { get; set; }
}

Task Add(string owner, string repo, string user, CollaboratorUpdate update);
```

**Comment by Haacked** [comment]
@ryangribble you're exactly right, it should be a request object. I was being lazy with my words. :smile:

**Comment by ryangribble** [comment]
Sweet, my understanding remains intact 😁

**Comment by ryangribble** [comment]
Also note that as part of this GitHub API change, they indicated that the `permission` parameter when creating and editing Teams is deprecated.  So potentially on our `NewTeam` and `UpdateTeam` request model classes, we should mark that field as `Obsolete`

**Comment by shiftkey** [comment]
> So potentially on our `NewTeam` and `UpdateTeam` request model classes, we should mark that field as Obsolete

👍

**Comment by ryangribble** [comment]
This was implemented in #1410 but I will raise a new issue to cover this other comment:

> Also note that as part of this GitHub API change, they indicated that the permission parameter when creating and editing Teams is deprecated. So potentially on our NewTeam and UpdateTeam request model classes, we should mark that field as Obsolete
